tests/phpunit/DataValueTest.php

```diff
@@ -19,7 +19,7 @@
  * @license GPL-2.0+
  * @author Jeroen De Dauw < jeroendedauw@gmail.com >
  */
-abstract class DataValueTest extends PHPUnit_Framework_TestCase {
+abstract class DataValueTest extends TestCase {
```

Comment:
Uh... this could also break tests in other packages because the used PHPunit version is only in require-dev. Shitty design is very shitty.

Reply:
Unfortunately, there's not much can be done about it besides quickly updating the other libraries once the version with this change is released. Which was in my plans from the beginning.

Reply:
A major release will save you here. It has to be a major release anyway, so I don't see the problem

Reply:
@AlaaSarhan its not just one major release. It is many. See https://www.entropywins.wtf/blog/2018/08/30/base-libraries-should-be-stable/. Its easy to create a lot of work for yourself with breaking changes in this library.

Reply:
sure it is many.. iwn't that what they arw desinged for.

Notice I am not arguing in regards of whether this change is a good idea or not. I do not know the full atory behind it. I'm just pointing out that it must be a major release, should it go through.

Reply:
Let me see if I get your point. You auggest that these changes are better not done because they require a major release, which turns out costly.

The question then, can we eo the version drops in this change without changing this class parent? I don't know yet, but if possible, I might agree in thepry tbat they are better left out to avoid a major release atm.

Reply:
I'm pointing out that major releases of this library are expensive (and mostly hidden if you don't think about what happens outside of this library).

This means that certain types of breaking changes in this library are not worth it and that batching breaking changes can be very worth it. So I'd like you to ask yourself the questions (a) if the break is really needed and (b) if it is needed, are there other breaking changes to batch? If another major release is needed, maybe it can be the last one.

Bumping PHP version does not require a major release. I do not agree the root cause has anything to do with update frequency. The cost of release in this case comes from impacted intermediate libraries that need their own releases, not from actual changes needed there or from the amount of these changes.

For a major release here to actually make it all the way to say Wikibase, someone has to do all the dull release work in the intermediate libraries. Last time that did not happen for months until I got to it, and that's actually what triggered me to write the blog post that I linked.

Reply:
More specifically: if the PHPUnit base class change is indeed breaking for consumers, then perhaps we can remove the DataValueTest class altogether since it is bad design to begin with and that way we avoid running into this situation again. Ideally we can avoid the break altogether and save us the work but I'm not sure that is possible.

Reply:
The specifity is very much appreciated.. we gotta stsrtore specific next time from the beginnig ;)

I agree to the reply before.. badically it is more elaborate version of what I was trying to say as well (failure of beingnspecific is always expensive)

I might still disagree slightly with the upgrade frequency as one root cause.. but iisn't worth that discussion on this thread.. for later times.

As for whether the change is needed. I think the versions upgrade is needed as staying up-to-date with supported versions is in general a good idea. This does entail amount of work repeatedly on thr chain of dependencies, but I also think it's worth it. How to reduce that amount of work is a good challenge. Your suggestion of keeping libs at the base "stable" is one way. There could br others too, esp. ones that do not compromise on following the (even more) base layer on which they are built on, such as a programing language in this example.

Reply:
I agree that this is fine, and it's what breaking changes are for. Let's get it done?